### PR TITLE
fix(#112): replace map by timestamp and add a getUser method

### PR DESCRIPTION
**Pull request by abouquet27**
This issue will need to migrate the database because it changes a component of the UserData class

### CODE REVIEW

File: app/src/main/java/com/github/studydistractor/sdp/ui/CreateUserScreen.kt [hunk 1]
```diff
@@ -190,6 +196,6 @@ fun CreateUserScreen(
 }
 
 private fun displayError(context: Context, data: String){
-    Log.d(Constants.TAG, "User has not been created $data")
+    Log.d("Create User", "User has not been created $data")
```
Comment (AthenaPK) [inline app/src/main/java/com/github/studydistractor/sdp/ui/CreateUserScreen.kt:199]: I would say keep the tag in the constants, it's cleaner like that.

File: app/src/main/java/com/github/studydistractor/sdp/ui/CreateUserScreen.kt [hunk 2]
```diff
@@ -163,9 +164,14 @@ fun CreateUserScreen(
         // Validate
         Button( onClick = {
             var isUserCreated = true
-            val birthday = hashMapOf(
-                "year" to year, "month" to month, "day" to dayOfMonth)
-            val user = UserData(userId, firstname, lastname, phone, birthday, 0)
+            val stmp = if (selectedDateText.isNotEmpty()) {
```
Comment (AthenaPK) [inline app/src/main/java/com/github/studydistractor/sdp/ui/CreateUserScreen.kt:169]: I think it's better if you do the date validation in the UserViewModel, i.e. just pass the birthday string to UserData and then checkBirthdayFormat from the viewmodel could be something like checkBirthdayFormatAndConvertToTimestamp. That way, the logic of validation will be in the viewmodel where it belongs.

File: app/src/androidTest/java/com/github/studydistractor/sdp/ui/CreateAccountScreenTest.kt
Comment (aizensoosuke) [inline app/src/androidTest/java/com/github/studydistractor/sdp/ui/CreateAccountScreenTest.kt:1]: Empty file ?

**Review by aizensoosuke — CHANGES_REQUESTED** [review]
Great work :) Thanks

File: app/src/androidTest/java/com/github/studydistractor/sdp/ui/CreateAccountScreenTest.kt
Comment (aizensoosuke) [reply app/src/androidTest/java/com/github/studydistractor/sdp/ui/CreateAccountScreenTest.kt:1]: Actually removed

**Review by aizensoosuke — DISMISSED** [review]

**Comment by abouquet27** [comment]
Failed to merge as there were problems in the rebase with the main branch

**Pull request closed without merging**